### DISCUSSION TITLE
How to use gdb for debugging?

### DISCUSSION THREAD

**Discussion by Shiva-Gantha** [aligungr/UERANSIM#407] [Q&A] [answered]
Hi,

I am new to UERANSIM, and I would like to use gdb for debugging. When I run gdb using the command:
`sudo gdb nr-ue`
and add try to add a break statement, it says "No symbol table is loaded."

I have added the following line in the makefile and recompiled:
`CFLAGS= -g -Wall -Wextra` as mentioned [here](https://stackoverflow.com/a/15265111/12504880)

But it still says "No symbol table added".

Please guide me on how to use gdb with UERANSIM.

Thanks!

**Comment by Shiva-Gantha** [comment]
I figured it out.
In the /cmake-build-release/CMakeCache.txt
I changed the `CMAKE_CXX_FLAGS:STRING=-g`, so that by default all the builds can be debugged using gdb

I understand there is a way to build UERANSIM in debug mode, and if we do that, this above change will not be required. But I could not find how to do so.

**Comment by aligungr** [comment, marked as answer]
You can also build UERANSIM in debug mode. Please see our `makefile`. I commented out 2 lines for debug builds.

You can switch between debug and release modes by editing the makefile.

**Reply by Shiva-Gantha** [reply]
Yes. I see it. Thanks!
